aten/src/ATen/native/mkldnn/xpu/qlinear.cpp

```diff
@@ -222,9 +236,13 @@ Tensor q_linear_pointwise_binary_tensor(
 
   std::vector<int64_t> src_dims = {M, K};
   std::vector<int64_t> dst_dims = {M, N};
-  auto out_dtype =
-      output_dtype.has_value() ? output_dtype.value() : act.scalar_type();
-  Tensor qout = at::empty(dst_dims, act.options().dtype(out_dtype));
+  bool fp32_output = output_dtype.has_value() && (output_dtype == c10::kFloat);
+  bool bfloat16_output =
+      output_dtype.has_value() && (output_dtype == c10::kBFloat16);
+  auto dst_dtype = fp32_output
+      ? c10::kFloat
+      : (bfloat16_output ? c10::kBFloat16 : act.scalar_type());
+  Tensor qout = at::empty(dst_dims, act.options().dtype(dst_dtype));
```

Comment:
Pls. get the code block abstacted as a utility function

Reply:
thanks for suggestion, the util is added at https://github.com/pytorch/pytorch/pull/136753/files#diff-140c7439ba3aa754bb216ee42172e969466f19353b28ac7a45975aa1f3bc97a9R10. Similar change is also introduced in #135465